src/components/PopupWithForm.js

```diff
@@ -6,12 +6,23 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._popupFormInputs =
       this._popupElement.querySelectorAll(".modal__input");
+    this._popupButton = this._popupElement.querySelector(".modal__button");
     this._handleFormSubmit = handleFormSubmit;
     this._handleSubmitListener = (evt) => {
       evt.preventDefault();
+      this._popupButton.textContent = "Saving...";
       const inputValues = this._getInputValues();
-      this._handleFormSubmit(inputValues);
-      this.close();
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`